crates/dash_node_impl/src/lib.rs

```diff
@@ -32,18 +33,27 @@ pub fn run_with_nodejs_mnemnoics(path: &str, opt: OptLevel, initial_gc_threshold
 
 async fn run_inner_fallible(path: &str, opt: OptLevel, initial_gc_threshold: Option<usize>) -> anyhow::Result<()> {
     let path = Path::new(path);
-    if !path.is_dir() {
+    let package_state = if path.is_dir() {
         // TODO: make it also work with paths to files. need to adjust the execute_node_module call too,
         // since that needs a dir path
-        bail!("Node project path currently needs to be a directory");
-    }
+        process_package_json(path)?
+    } else {
+        PackageState {
+            base_dir: path.parent().unwrap_or(&env::current_dir()?).into(),
+            metadata: Package::default_with_entry(
+                String::from_utf8_lossy(path.to_str().unwrap_or_default().as_bytes()).to_string(),
```

Comment:
if you move the `let path = Path::new(path)` into the other if branch, then we can simplify this long line to just `path.to_owned()`

Reply:
actually, it would be even better if the `entry` field in the `Package` struct was a `PathBuf` directly. then this could also simply be `path.to_owned()`